boostsec/converter/sarif/mobsfscan_sast.py

```diff
@@ -0,0 +1,273 @@
+"""Boost SAST extensions."""
+from typing import Optional, TypedDict
+
+from .base import (
+    ReportingDescriptor,
+    ReportingDescriptorReference,
+    ReportingDescriptorRelationship,
+    Run,
+    ToolComponent,
+    ToolComponentReference,
+)
+from .boostsec import BoostFindingConfidence, BoostFindingSeverity
+
+BOOST_SAST_TAXONOMY_NAME = "boost/sast"
+BOOST_SAST_TAXONOMY_VERSION = "1.0.0"
+BOOST_SAST_TAXONOMY_ORG = "boostsecurity"
+
+
+class MobsfscanCweType(TypedDict):
+    """Represents a typed dict for a given CWE."""
+
+    cwe_id: str
+    cwe_title: str
+    severity: BoostFindingSeverity
+    confidence: BoostFindingConfidence
+
+
+DEFAULT_RULE = MobsfscanCweType(
+    confidence=BoostFindingConfidence.NOT_SET,
+    cwe_id="CWE-UNKNOWN",
+    cwe_title="CWE-UNKNOWN - Original rule did not map to a known rule",
+    severity=BoostFindingSeverity.NOT_SET,
+)
+
+DEFAULT_RULE_KEY = DEFAULT_RULE.get("cwe_id")
+assert DEFAULT_RULE_KEY  # noqa S101 # type hint
+
+CWE_ALLOW_LIST: dict[str, MobsfscanCweType] = {
+    "CWE-73": {
```

Comment:
by using the mitre rules db with the default, there won't be a need for CWE allow list.

Reply:
Similar to earlier comment, should I remove occurrences of `CWE_ALLOW_LIST` and it's usage?

Reply:
that's right. The CWE_ALLOW_LIST is essentially the list of CWE Ids that exist in the rules db on the server. The book keeping used to be done that way so that the converter can know the CWE is unknown in the server and the converter sends a default in that case. Since the default can now be managed on the server, there is no need for book keeping and default in the converter.